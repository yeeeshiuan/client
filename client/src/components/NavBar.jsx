import React from 'react';
import 'react-bulma-components/dist/react-bulma-components.min.css';
import { Navbar } from 'react-bulma-components';


const NavBar = (props) => (
    <Navbar color="dark">
      <Navbar.Brand>
        <Navbar.Item renderAs="div" href="#">
            <Navbar.Item href="/">{props.title}</Navbar.Item>
        </Navbar.Item>
        <Navbar.Burger/>
      </Navbar.Brand>
      <Navbar.Menu>
        <Navbar.Container>
          <Navbar.Item href="/about">About</Navbar.Item>
        </Navbar.Container>
        <Navbar.Container position="end">
          <Navbar.Item href="#">At the end</Navbar.Item>
        </Navbar.Container>
      </Navbar.Menu>
    </Navbar>
)

export default NavBar;
