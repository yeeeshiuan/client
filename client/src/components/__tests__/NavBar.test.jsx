import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import NavBar from '../NavBar';


test('NavBar renders properly', () => {
  const title = 'Hello, World!';
  const wrapper = shallow(<NavBar title={title}/>);
  //console.log(wrapper.debug());
  const element = wrapper.find('d');
  expect(element.length).toBe(6);
  expect(element.get(1).props.children).toBe(title);
  expect(element.get(3).props.children).toBe('About');
  expect(element.get(5).props.children).toBe('At the end');
});

test('NavBar renders a snapshot properly', () => {
  const tree = renderer.create(
    <Router location="/"><NavBar/></Router>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
