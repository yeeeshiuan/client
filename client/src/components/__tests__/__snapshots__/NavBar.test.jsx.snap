// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders a snapshot properly 1`] = `
<nav
  className="navbar is-dark"
  role="navigation"
  style={Object {}}
>
  <div
    className="navbar-brand"
    style={Object {}}
  >
    <div
      className="navbar-item"
      href="#"
      style={Object {}}
    >
      <a
        className="navbar-item"
        href="/"
        style={Object {}}
      />
    </div>
    <div
      className="navbar-burger"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="0"
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu"
    style={Object {}}
  >
    <div
      className="navbar-start"
      style={Object {}}
    >
      <a
        className="navbar-item"
        href="/about"
        style={Object {}}
      >
        About
      </a>
    </div>
    <div
      className="navbar-end"
      style={Object {}}
    >
      <a
        className="navbar-item"
        href="#"
        style={Object {}}
      >
        At the end
      </a>
    </div>
  </div>
</nav>
`;
